api/edge_api/identities/models.py

```diff
@@ -200,23 +196,21 @@ def to_document(self) -> dict:
         return map_engine_identity_to_identity_document(self._engine_identity_model)
 
     def _update_feature_overrides(
-        self,
-        changeset: IdentityChangeset,
-        user: FFAdminUser = None,
-        master_api_key: MasterAPIKey = None,
+        self, changeset: IdentityChangeset, user: FFAdminUser | APIKeyUser
     ) -> None:
         if changeset["feature_overrides"]:
             # TODO: would this be simpler if we put a wrapper around FeatureStateModel instead?
-            generate_audit_log_records.delay(
-                kwargs={
-                    "environment_api_key": self.environment_api_key,
-                    "identifier": self.identifier,
-                    "user_id": getattr(user, "id", None),
-                    "changes": changeset,
-                    "identity_uuid": str(self.identity_uuid),
-                    "master_api_key_id": getattr(master_api_key, "id", None),
-                }
-            )
+            kwargs = {
+                "environment_api_key": self.environment_api_key,
+                "identifier": self.identifier,
+                "user_id": getattr(user, "id", None),
```

Comment:
This will start to fail in the future if someone adds the `id` attribute to the `APIKeyUser` class. Will a test pick it up if so? Or should we be more explicit with this check? e.g. 

```
"user_id": user.id if not getattr(user, "is_master_api_key_user", False) else None
```